plugins/out_kinesis_firehose/firehose_api.c

```diff
@@ -216,12 +217,13 @@ static int process_event(struct flb_firehose *ctx, struct flush *buf,
                          ctx->delivery_stream);
             return 2;
         }
-        /* guess space needed to write time_key */
-        len = 6 + strlen(ctx->time_key) + 6 * strlen(ctx->time_key_format);
+        /* format time output and get the length */
+        len = create_time_formatter(&out_buf, ctx->time_key_format, tms);
         /* how much space do we have left */
         tmp_size = (buf->tmp_buf_size - buf->tmp_buf_offset) - written;
-        if (len > tmp_size) {
-            /* not enough space- tell caller to retry */
+        if (len > tmp_size && len == -1) {
+            /* not enough space or function create_time_formatter return -1 - tell caller to retry */
```

Comment:
Please read this document: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#line-length

No line should exceed 90 characters in length